das-jdbc/src/main/java/org/wso2/das/jdbcdriver/common/SQLParser.java

```diff
@@ -0,0 +1,215 @@
+import org.gibello.zql.ZSelectItem;
+import org.gibello.zql.ZStatement;
+import org.gibello.zql.ZqlParser;
+import org.wso2.das.jdbcdriver.aggregateFunctions.MaxAggrFunction;
+import org.wso2.das.jdbcdriver.aggregateFunctions.MinAggrFunciton;
+import org.wso2.das.jdbcdriver.aggregateFunctions.SumAggrFunction;
+import org.wso2.das.jdbcdriver.aggregateFunctions.CountAggrFunction;
+import org.wso2.das.jdbcdriver.expressions.AndExpression;
+import org.wso2.das.jdbcdriver.expressions.AsteriskExpression;
+import org.wso2.das.jdbcdriver.expressions.ColumnName;
+import org.wso2.das.jdbcdriver.expressions.Expression;
+import org.wso2.das.jdbcdriver.expressions.GeneralExpression;
+import org.wso2.das.jdbcdriver.expressions.ORExpression;
+import org.wso2.das.jdbcdriver.expressions.RelationOpExpression;
+
+import java.io.ByteArrayInputStream;
+import java.io.DataInputStream;
+import java.io.InputStream;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Vector;
+
+/**
+ * This class have the functions required for parsing the SQL. ZQL Sql parser is used for this.
+ */
+public class SQLParser {
+
+    private List<Object[]> queryEnvironment = new ArrayList<Object[]>();
+    private Expression whereExpression;
+    private String sTableName;
+    private String sql;
+
+    public SQLParser(String sql) {
+        this.sql = sql;
+    }
+
+    /**
+     * Parse the SQL Query String by using the ZQL parser.
+     */
+    public void parse() throws SQLException {
+        System.out.println("SQL:" + sql);
+        if (sql.length() > 0) {
+            if (sql.charAt(sql.length() - 1) != ';') {
+                sql = sql.concat(";");
+            }
+        }
+        InputStream is = new ByteArrayInputStream(sql.getBytes());
+        ZqlParser p = new ZqlParser(new DataInputStream(is));
+        ZStatement st = null;
+        try {
+            st = p.readStatement();
+        } catch (ParseException e) {
+            throw new SQLException("Error in Parsing SQL:",e);
+        }
+        if (st != null) {
+            if (st instanceof ZQuery) {
+                ZQuery q = (ZQuery) st;
+                Vector sel = q.getSelect(); // SELECT part of the query
+                Vector from = q.getFrom();  // FROM part of the query
+                ZExpression where = (ZExpression) q.getWhere();  // WHERE part of the query
+                if (where != null) {
+                    this.whereExpression = getWhereClauseExpression(where);
+                }
+                if (from.size() > 1) {
+                    throw new SQLException("Joins are not supported");
+                }
+                // Retrieve the table name in the FROM clause
+                ZFromItem table = (ZFromItem) from.elementAt(0);
+                this.sTableName = table.getTable();
+                if (sel.size() == 0) {
+                    this.queryEnvironment.add(new Object[] { "*", new AsteriskExpression("*") });
```

Comment:
Better to define asterisk (*) and other constant literals as constants and use them